.github/workflows/publish-and-sync-repos.yml

```diff
@@ -39,3 +40,29 @@ jobs:
         run: ./scripts/publish.sh
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+          
+  check_changes_and_update_repos:
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+    name: Code Changes
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: write
+      pull-requests: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set script permissions
+        run: |
+          chmod +x ./scripts/check-changes.sh
```

Comment:
We should fix this in the git repo.

https://stackoverflow.com/questions/40978921/how-to-add-chmod-permissions-to-file-in-git